src/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -107,3 +109,37 @@ std::set<RobotCapability> &Tactic::mutableRobotCapabilityRequirements()
 {
     return capability_reqs;
 }
+
+std::unique_ptr<Intent> Tactic::get(const Robot &robot, const World &world)
+{
+    // TODO (#1888): remove these updates
+    updateWorldParams(world);
+    updateRobot(robot);
+
+    updateIntent(TacticUpdate{.robot      = robot,
+                              .world      = world,
+                              .set_intent = [this](std::unique_ptr<Intent> new_intent) {
+                                  intent = std::move(new_intent);
+                              }});
+    if (intent)
+    {
+        return std::move(intent);
+    }
+    else
+    {
+        LOG(WARNING) << "No intent set for this tactic" << std::endl;
+        return std::make_unique<StopIntent>(robot.id(), false);
+    }
+}
+
+void Tactic::updateIntent(const TacticUpdate &tactic_update)
+{
+    // Try to get an intent from the tactic
+    std::shared_ptr<Action> action = getNextAction();
```

Comment:
Is this action stuff just to not break compatibility w/ other tactics? will this function will go away with #1888?

Reply:
yes